sdk/src/Core/Amazon.Runtime/Internal/Transform/JsonUnmarshallerContext.cs

```diff
@@ -88,13 +89,22 @@ public JsonUnmarshallerContext(
             bool isException,
             IRequestContext requestContext)
         {
+            // In the case of an exception or when maintainResponseBody is true, we must maintain a copy of the stream.
+            // This is because during unmarshalling, a new StreamingUtf8JsonReader is instantiated and in its constructor the stream is automatically read
+            // to fill the buffer. This advances the position of the stream to the end and subsequent calls to Context.ResponseBodyBytes will return a 0 byte array 
+            // since there is nothing to read. To avoid this, we maintain a copy of the stream if an exception occurs or if maintainResponseBody is true.
+
             if (isException)
             {
-                this.WrappingStream = new CachingWrapperStream(responseStream);
+                var copyStream = new MemoryStream();
+                AWSSDKUtils.CopyStreamAndSetPositionToZero(responseStream, copyStream);
+                this.WrappingStream = new CachingWrapperStream(copyStream);
             }
             else if (maintainResponseBody)
             {
-                this.WrappingStream = new CachingWrapperStream(responseStream, AWSConfigs.LoggingConfig.LogResponsesSizeLimit);
+                var copyStream = new MemoryStream();
+                AWSSDKUtils.CopyStreamAndSetPositionToZero(responseStream, copyStream);
```

Comment:
The point of the `AWSConfigs.LoggingConfig.LogResponsesSizeLimit` is we don't buffer too large of a response in memory as we stream the response into the marshaller. By coping the entire response into a separate `MemoryStream` we make the `AWSConfigs.LoggingConfig.LogResponsesSizeLimit` useless. Actually worse because we have the entire response buffered in the `MemoryStream` and then the cache in the `CachingWrapperStream` up to `AWSConfigs.LoggingConfig.LogResponsesSizeLimit`.